src/components/common/animation-control/animation-controller.ts

```diff
@@ -47,14 +58,8 @@ function AnimationControllerFactory() {
    * Current time is a point. An array of sorted time steps are provided,
    * animate a moving point jumps to the next step
    */
-  class AnimationController extends Component {
+  class AnimationController extends Component<AnimationControllerProps> {
     // TODO: convert the entire component to use hooks in the next PR
-    static defaultProps = {
```

Comment:
is there a reason we can't keep `defaultProps`? I find it a lot more sane then adding values wherever we deconstruct `this.props`

Reply:
The problem of defaultProps is that typescript doesn't see it. So sometimes I should duplicate it with deconstruction default values... But I'll keep it for the cases where it's not necessary